pandas/io/formats/format.py

```diff
@@ -1126,39 +1122,22 @@ def _format_strings(self):
         return fmt_values.tolist()
 
 
-class IntervalArrayFormatter(GenericArrayFormatter):
-
-    def __init__(self, values, *args, **kwargs):
-        GenericArrayFormatter.__init__(self, values, *args, **kwargs)
-
-    def _format_strings(self):
-        formatter = self.formatter or str
-        fmt_values = np.array([formatter(x) for x in self.values])
-        return fmt_values
-
-
-class PeriodArrayFormatter(IntArrayFormatter):
-
+class ExtensionArrayFormatter(GenericArrayFormatter):
     def _format_strings(self):
-        from pandas.core.indexes.period import IncompatibleFrequency
-        try:
-            values = PeriodIndex(self.values).to_native_types()
-        except IncompatibleFrequency:
-            # periods may contains different freq
-            values = Index(self.values, dtype='object').to_native_types()
-
-        formatter = self.formatter or (lambda x: '{x}'.format(x=x))
```

Comment:
I *think* this is the cause of the extra space. Periods were incorrectly missing a space earlier.

On 0.23.4, notice that the period values start above the `e` in `dtype`, rather than above the `:` like the rest:

```python
In [9]: pd.Series(range(3))
Out[9]:
0    0
1    1
2    2
dtype: int64

In [10]: pd.Series(pd.period_range('2000', periods=3))
Out[10]:
0   2000-01-01
1   2000-01-02
2   2000-01-03
dtype: object

```

On this PR, periods start above the colon:

```python
In [7]: pd.Series(range(3))
Out[7]:
0    0
1    1
2    2
dtype: int64

In [8]: pd.Series(pd.period_range('2000', periods=3))
Out[8]:
0    2000-01-01
1    2000-01-02
2    2000-01-03
dtype: period[D]
```